src/lib/bridge/handlers/fetch_recommended_price.ts

```diff
@@ -0,0 +1,97 @@
+import {SimpleHandler} from './main';
+import {RequestType} from './types';
+import {environment} from '../../../environment';
+import MessageSender = chrome.runtime.MessageSender;
+
+export interface CSFloatListing {
+    id: string;
+    created_at: string;
+    type: 'buy_now' | 'auction';
+    price: number; // in cents
+    state: string;
+    item: {
+        market_hash_name: string;
+    };
+}
+
+export interface FetchRecommendedPriceRequest {
+    market_hash_name: string;
+}
+
+export interface FetchRecommendedPriceResponse {
+    price: number;
+}
+
+const MIN_LISTINGS_FOR_AVERAGE = 3;
+const MAX_LISTINGS_TO_FETCH = 50;
+
+async function fetchRecommendedPrice(
+    req: FetchRecommendedPriceRequest,
+    sender: MessageSender
+): Promise<FetchRecommendedPriceResponse> {
+    // Build query parameters
+    const params: Record<string, string> = {
+        market_hash_name: req.market_hash_name,
+        limit: MAX_LISTINGS_TO_FETCH.toString(),
+        sort_by: 'lowest_price',
+        type: 'buy_now',
+    };
+
+    const url = `${environment.csfloat_base_api_url}/v1/listings?${new URLSearchParams(params)}`;
```

Comment:
Generally we want to avoid "expensive" listing calls like this -- this endpoint has a ~200/hour rate limit right now AFAIK and it's common for users to both list items and search them. More likely than not, they hit the rate limit.

If we mostly want to get the current minimum price for the item by market hash name, then you can use `https://csfloat.com/api/v1/listings/price-list` and cache it internally for an hour or two.